### DISCUSSION TITLE
suppressing kj Warnings

### DISCUSSION THREAD

**Discussion by kentdlee** [capnproto/capnproto#1969] [Q&A]
I am seeing the following printed while running code that has CapnProto linked into it. 

kj/filesystem-disk-unix.c++:1703: warning: PWD environment variable doesn't match current directory

I'd like to suppress warning messages like this since it really isn't a relevant message to how we are using the library. Is there a compile flag I can set to suppress these messages? I understand it is a log message. I would like them not printed while running if that's possible. Not sure how to turn that off.

**Comment by kentonv** [comment]
See #1892 for a discussion of this warning. Per the discussion I'm amenable to getting rid of it.

If you want to filter or silence KJ logging in general, you can do so using a kj::ExceptionCallback.

**Comment by kentdlee** [comment]
I am seeing this from the pycapnp library. It is the code printing the warnings. If I wanted to turn off those warnings is there a way to do this? I tried suppressing stdout, stderr from within Python but that didn't seem to work.

**Reply by kentonv** [reply]
I don't know if pycapnp provides a way to do this. Maybe it should. You'd have to ask the pycapnp maintainer about that.

**Comment by kentdlee** [comment]
That same warning is printing for a colleague of mine while running 

capnp compile -oc++ message_defs.capnp

in a make file while building in a container. It would be nice if we could eliminate the printing of it and if making it an info level message is the way to go I am all for it. Do you want to make that change, Kenton? Should I record an issue? Let me know what you would like or I can make the change and put up a PR if you like.

**Reply by kentdlee** [reply]
I guess I am assuming that making a change in the capnproto repo will also result in the same change in the pycapnp project since it is a wrapper to the other.

**Reply by kentonv** [reply]
Please send a PR (probably against the `master` branch, which represents the 1.x releases; not `v2` which is the experimental development branch).

You will then probably need to rebuild pycapnp against your updated C++ library. Sorry, I don't know how exactly that is done.

**Reply by kentdlee** [reply]
Sounds good. Thanks!

**Reply by kentdlee** [reply]
I have submitted the PR #1977. Thanks!
